File: WeeklyChallengeOne-master/app/src/main/java/com/slack/weeklychallengeone/Utils/MyAdapter.java
package com.slack.weeklychallengeone.Utils;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.slack.weeklychallengeone.R;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    private ArrayList<Model> mArrayList;


    public MyAdapter(ArrayList<Model> mArrayList) {
        this.mArrayList = mArrayList;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.row_layout, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.setIsRecyclable(false);
        holder.tv_name.setText(mArrayList.get(position).getName());
        holder.cardView.setBackgroundResource(mArrayList.get(position).getBackgroundCard());

    }

    @Override
    public int getItemCount() {
        return mArrayList.size();
    }


    class MyViewHolder extends RecyclerView.ViewHolder{
        private TextView tv_name;
        private ImageView cardView;
        MyViewHolder(View view) {
            super(view);

            tv_name = view.findViewById(R.id.tv_name);
            cardView = view.findViewById(R.id.card_bg);

        }
    }


}

File: WeeklyChallengeOne-master/app/src/main/java/com/slack/weeklychallengeone/utils/MyAdapter.kt
package com.slack.weeklychallengeone.utils

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.slack.weeklychallengeone.R
import kotlinx.android.synthetic.main.row_layout.view.*
import java.util.*

class MyAdapter(private val mArrayList: ArrayList<Model>) : RecyclerView.Adapter<MyAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.row_layout, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.name.text = mArrayList[position].name
        holder.description.text = mArrayList[position].description
        holder.foldingCell.setOnClickListener {
            holder.foldingCell.toggle(false)
        }
    }

    override fun getItemCount(): Int {
        return mArrayList.size
    }


    inner class MyViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var name = itemView.tv_name!!
        var description = itemView.tv_description!!
        var foldingCell = itemView.folding_cell!!
    }
}
